### PR TITLE
chore(ci): automate BDD tests in CI pipeline

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this pull request do?

It adds shell scripts that enable the execution of Cucumber tests, supporting two modes:

1. local mode, with the "run-bdd" script, which uses local dependencies to run the tests (it requires "bundle install")
2. docker mode, with the `features.sh` script, which runs the tests inside a Ruby container

We are also adding a stage on Jenkins for the BDD tests, using the dockerized version of the automation. This stage wont break the build if they fail. As this moment, they run just for latest ruby version (2.7), so we probably need to define if it's convenient to run them for more versions.

Finally, we renamed a variable in a shell script for consistency.

## Why is it important?
Include test coverage in the CI pipeline, adding visibility about existing tests.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). 
- [x] My code follows the style guidelines of this project (See `.rubocop.yml`)
- [x] I have rebased my changes on top of the latest master branch

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-ruby/blob/master/CONTRIBUTING.md#testing) pass locally with my changes

- [x] I have made corresponding changes to the documentation
- [ ] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
- [ ] I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
- [ ] Added an API method or config option? Document in which version this will be introduced

### CODE REVIEW

**Review by cachedout — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -143,6 +143,34 @@ pipeline {
         }
       }
     }
+    stage('BDD Tests') {
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:146]: Hey @elastic/apm-agent-ruby team. This stage is not necessarily needed, although we see value automating it on CI side.

If so, we should define when it must be run: on PRs, on merges, etc.

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -143,6 +143,34 @@ pipeline {
...
+        HOME = "${env.WORKSPACE}"
+        PATH = "${env.WORKSPACE}/bin:${env.PATH}"
+      }
+      steps {
+        withGithubNotify(context: 'BDD Tests') {
+          deleteDir()
+          unstash 'source'
+          dir(BASE_DIR) {
+            catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE', message: 'BDD Tests failed but keep running the build') {
+              sh(label: "Run BDD tests", script: "./spec/scripts/features.sh ruby:2.7")
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:168]: As of now, it only runs for one Ruby version. Should it be run for all of them?

Comment (mikker) [reply .ci/Jenkinsfile:168]: Sure 👍

Comment (mdelapenya) [reply .ci/Jenkinsfile:168]: Then we can integrate it into the main matrix

Comment (mdelapenya) [reply .ci/Jenkinsfile]: I added 5880d66c5b1acbbaac736fb9b174c2832fe0f230 to run the BDD tests with every Ruby version, for each CI build. Do we want to run them on PRs too?

**Comment by mikker** [comment]
@mdelapenya Is this done or are you still working on it?

**Comment by mdelapenya** [comment]
> @mdelapenya Is this done or are you still working on it?

Sorry @mikker, returned from vacations past week and did not attend this PR yet. Yes, let's keep it as draft while I get context again.

At a quick sight, I see a lot of test errors, probably not-related.

**Comment by mikker** [comment]
@mdelapenya Is this good to go?

**Comment by mdelapenya** [comment]
Yes, ready to be reviewed and eventually merged, thanks!

**Pull request merged**